Automattic/wp-calypso
# 100 year plan: New/Existing site flow

Fixes Automattic/martech#2184
P2: pau2Xa-5bu-p2

## Proposed Changes

* This PR adds the new/existing site flow for the 100 year plan.

| Step | Desktop | Mobile |
|--------|--------|--------|
| New or Existing Site |<img width="1907" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/73b8a754-5892-4642-8b8a-8979b122de99"> | <img width="311" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/09d3ea83-7fa4-421e-b657-9ae2c441c83b"> |
| Site Picker |<img width="1907" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/6de761c5-6571-47dd-bb44-94d909a096a6"> | <img width="366" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/2787b14d-b377-4049-87f3-b2c861b4db5c"> |
| Site Picker Modal | <img width="1904" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/b3eea66e-4f5f-42ed-aa4e-5a9ad563b9f3"> | <img width="368" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/cb8069ba-5403-4f14-b013-7fd833780b01"> | 

## Testing Instructions



- Go to /setup/hundred-year-plan
- If the user has 0 existing sites:
  - The Site title/description step should be shown. The site title field is mandatory.)
  - The next step should be the domains step. The prices for the domains should be hidden. A *.wordpress.com domain suggestions should not be visible.
  - Selecting a domain should lead to checkout.
- If the user has 1 or more existing sites:
  - The New/Existing site step should be shown.
  - Clicking on the “New Site” CTA should start the new-site flow listed above.
  - Clicking on the “Existing Site” CTA should lead to the site picker step.
  - The site picker step should show a list of the user’s sites and should exclude <TBD>.
  - Selecting an existing site should show a confirmation modal.
  - Clicking on the "Chat with us" link should open up the chat window.
  - Clicking on the “Confirm” CTA in the modal should lead to checkout.
- The pre-Checkout loading screen should be according to the design (“Setting up your legacy…”).

## Pre-merge Checklist
